python-main.js

```diff
@@ -1225,9 +1228,7 @@ function web_editor(config) {
             // Push binary to board
             p = PartialFlashing.connectDapAsync()
                 .then(function() {
-                    var output = generateFullHexStr();
-                    var image = MemoryMap.fromHex(output).slicePad(0, PartialFlashingUtils.pageSize * PartialFlashingUtils.numPages);
-
+                    var image = generateFullHex("bytes");
```

Comment:
I just realised that `generateFullHex` throws the erros up, so we need to catch them here like it is done in the non-partial flashing path:

![image](https://user-images.githubusercontent.com/29712657/63952297-37f17200-ca77-11e9-950d-84c73e8afe32.png)

Reply:
Since this is inside a promise, we might have to do this before `p = PartialFlashing.connectDapAsync()`.

Reply:
Is there an easy way to trigger the error?

I've added a catch to the end of the promise which should capture it

```
+                .catch(function(e) {
+                    alert(config.translate.alerts.error + e.message);
+                    return;
+                });
```

Reply:
Added a comment in https://github.com/bbcmicrobit/PythonEditor/pull/225#discussion_r319142672.

The way I triggered the error was by modifying generateFullHex() and added a `threw new Error('test')` as the first line.